### ISSUE TITLE
MPNet: Masked and Permuted Pre-training for Language Understanding

### ISSUE THREAD

**Issue by ymgw55** [ymgw55/MLPapers#52]
# Information
### URL
https://arxiv.org/abs/2004.09297
### Published at
Microsoft
### Date of issue
20 Apr 2020 

# Reading Motivation
最近のモデルについて学ぶ
# Summary
* BERT
masked language model (MLM). [mask] 同士の依存関係を無視している.
* XLNet
permuted language model (PLM). BERTの[mask]の依存関係の問題は解決したが, もとの文章の順番の情報を有効に使えていないため, pre-training と fine-tuning で token の位置情報に乖離がある. (例えばpermuteされた文章中のまだ見ていないtokenの位置情報)

MPNet は BERTとXLNetのいいとこ取りをして上の問題を解消したモデル.

# Background
* MLMもPLMもマスクされてない部分とマスクされていて予測される部分の2つに分けることができる.
![Screenshot from 2021-05-22 05-45-50](https://user-images.githubusercontent.com/39965001/119196381-06c59a80-bac1-11eb-8e7c-1beca5e683ad.png)
* これによりMLMとPLMは確率の条件は異なるが, 似た形で書き直すことができるとわかる.
![Screenshot from 2021-05-22 05-47-50](https://user-images.githubusercontent.com/39965001/119196538-4b513600-bac1-11eb-93cc-e8f7d17aed2b.png)

# Proposed Method
Backgroundを踏まえて, MPNetのObjectiveを以下で定義する
![Screenshot from 2021-05-22 05-50-34](https://user-images.githubusercontent.com/39965001/119196759-aaaf4600-bac1-11eb-91d8-d2b5286cc0fb.png)
permute された順番を扱うのでXLNet 同様, two-stream self attention を用いる.
XLNetと異なり, query stream が残りのmask token, 及びそれらの位置情報を持つことができる. (図の理解が間違っているかも...)
![名称未設定のアートワーク](https://user-images.githubusercontent.com/39965001/119204982-dfc39480-bad1-11eb-8435-9808238d4b0b.png)

MLM, PLMとの違いは下が分かりやすい.
![Screenshot from 2021-05-22 07-59-48](https://user-images.githubusercontent.com/39965001/119205653-c58ab600-bad3-11eb-9e28-3eb0828b0901.png)
